### PR TITLE
[FORWARDPORT] Make comparators able to act on keys and values

### PR DESCRIPTION
**Pull request by taburet**
A custom paging predicate comparator may act on keys and values at the
same time even if only the keys were requested, e.g. using `IMap.keySet`
call. Before this fix only keys were fetched for `IMap.keySet`, making
comparators unable to act on values.

Original maintenance PR: https://github.com/hazelcast/hazelcast/pull/15324

### CODE REVIEW

**Comment by mmedenjak** [comment]
@taburet I think we can merge this with a single approval. WDYT?

**Comment by taburet** [comment]
@mmedenjak it's better to review it, it's not a clean cherry-pick, mostly because the client code was merged into the core.

File: hazelcast/src/test/java/com/hazelcast/client/map/ClientPagingPredicateTest.java [hunk 1]
```diff
@@ -444,6 +445,32 @@ public void testLargePageSizeIsNotCausingIndexOutBoundsExceptions() {
         }
     }
 
+    @Test
+    public void testCustomComparatorAbleToActOnKeysAndValues() {
```
Comment (petrpleshachkov) [inline hazelcast/src/test/java/com/hazelcast/client/map/ClientPagingPredicateTest.java]: It seems in the `PagingPredicateTest` we have exactly the same test. Can we somehow avoid code duplication?

Comment (taburet) [reply hazelcast/src/test/java/com/hazelcast/client/map/ClientPagingPredicateTest.java]: Merged member and client paging predicate tests together.

**Pull request merged**